src/openapi-sampler.js

```diff
@@ -11,7 +19,14 @@ const defaults = {
 export function sample(schema, options, spec) {
   let opts = Object.assign({}, defaults, options);
   clearCache();
-  return traverse(schema, opts, spec).value;
+  let result = traverse(schema, opts, spec).value;
+  if (opts.format === 'xml') {
+    if (Object.keys(result).length > 1) {
```

Comment:
add some comment explaining this

Reply:
also, what if result is string or null. I would add some checks here and throw some meaningful error if it happens (which we catch in ref docs and use some fallback value instead of creashing whole page)

Reply:
added comments and moved generating xml logic to separate function.